doc/provenance.md

```diff
@@ -0,0 +1,80 @@
+<!--
+SPDX-FileCopyrightText: 2024 Technology Innovation Institute (TII)
+
+SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Getting Started
+
+To get started, follow the [Getting Started](../README.md#getting-started) section from the main [README](../README.md).
+
+As an example, to run the [`provenance`](../src/provenance/main.py) tool from the `tiiuae/sbomnix` repository:
+
+```bash
+# '--' signifies the end of argument list for `nix`.
+# '--help' is the first argument to `provenance`
+nix run github:tiiuae/sbomnix#provenance -- --help
+```
+
+# provenance
+
+[`provenance`](../src/provenance/main.py) is a command line tool to generate SLSA v1 compliant provenance attestation files in json format for any nix flake or derivation.
+
+To generate provenance file for `nixpkgs#hello`:
+
+```bash
+provenance nixpkgs#hello
+```
+
+To generate provenance file for `curl-8.6.0` in your nix store:
+
+```bash
+provenance /nix/store/fh7vxc5xgiwl6z7vwq5c3lj84mpcs4br-curl-8.6.0-bin
+```
+
+By default the dependecies are resolved only at the top level. ie. only direct dependecies.
+To get all dependencies recursively, you can use the `--recursive` option.
+Note the this will result in a very long provenance file.
```

Comment:
I would add a mention somewhere here that the dependencies now included are buildtime depdendencies.
I don't know if SLSA specs actually state if the provenance should include buildtime or only runtime dependencies.

Not necessary a topic of this PR, but if you are going to add a support to outputting only the runtime dependencies, you should align the command-line usage with other tools in this repo, that is, default to runtime dependencies with the `--buildtime` option changing the output to include also buildtime dependencies.

Reply:
from [SLSA spec](https://slsa.dev/spec/v1.0/provenance): 
> All artifacts fetched during initialization or execution of the build process are considered dependencies

To me this sounds like only buildtime dependencies are relevant.